diff -upr Src/main.c Src/main.c
--- Src/main.c	2021-04-20 14:50:08.129003300 +0900
+++ Src/main.c	2021-04-20 11:09:32.000000000 +0900
@@ -24,7 +24,7 @@
 
 /* Private includes ----------------------------------------------------------*/
 /* USER CODE BEGIN Includes */
-
+#include "main_standalone_mode_sample_app.h"
 /* USER CODE END Includes */
 
 /* Private typedef -----------------------------------------------------------*/
@@ -89,7 +89,7 @@ int main(void)
   MX_USART1_UART_Init();
   MX_USART2_UART_Init();
   /* USER CODE BEGIN 2 */
-
+  main_standalone_mode_sample_app();
   /* USER CODE END 2 */
 
   /* Infinite loop */
diff -upr Src/stm32l0xx_it.c Src/stm32l0xx_it.c
--- Src/stm32l0xx_it.c	2021-04-20 14:50:07.686468100 +0900
+++ Src/stm32l0xx_it.c	2021-04-20 11:09:32.000000000 +0900
@@ -23,6 +23,7 @@
 #include "stm32l0xx_it.h"
 /* Private includes ----------------------------------------------------------*/
 /* USER CODE BEGIN Includes */
+#include "CXM150x_Port.h"
 /* USER CODE END Includes */
 
 /* Private typedef -----------------------------------------------------------*/
@@ -145,7 +146,9 @@ void SysTick_Handler(void)
 void EXTI4_15_IRQHandler(void)
 {
   /* USER CODE BEGIN EXTI4_15_IRQn 0 */
-
+    if(__HAL_GPIO_EXTI_GET_IT(GPIO_PIN_10) != RESET){
+        wrapper_CXM150x_int_out2();
+    }
   /* USER CODE END EXTI4_15_IRQn 0 */
   HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_10);
   /* USER CODE BEGIN EXTI4_15_IRQn 1 */
diff -upr Src/usart.c Src/usart.c
--- Src/usart.c	2021-04-20 14:50:07.545171800 +0900
+++ Src/usart.c	2021-04-20 11:09:32.000000000 +0900
@@ -21,7 +21,7 @@
 #include "usart.h"
 
 /* USER CODE BEGIN 0 */
-
+extern void wrapper_CXM150x_uart_rx_callback(void);
 /* USER CODE END 0 */
 
 UART_HandleTypeDef huart1;
@@ -185,7 +185,11 @@ void HAL_UART_MspDeInit(UART_HandleTypeD
 }
 
 /* USER CODE BEGIN 1 */
-
+void HAL_UART_RxCpltCallback(UART_HandleTypeDef *UartHandle){
+    if(UartHandle->Instance==USART1){
+        wrapper_CXM150x_uart_rx_callback();
+    }
+}
 /* USER CODE END 1 */
 
 /************************ (C) COPYRIGHT STMicroelectronics *****END OF FILE****/
