diff --git a/tensorflow/lite/micro/tools/make/targets/spresense_makefile.inc b/tensorflow/lite/micro/tools/make/targets/spresense_makefile.inc
index 39363fbaadf..baa45e12909 100644
--- a/tensorflow/lite/micro/tools/make/targets/spresense_makefile.inc
+++ b/tensorflow/lite/micro/tools/make/targets/spresense_makefile.inc
@@ -88,7 +88,7 @@ THIRD_PARTY_CC_SRCS := \
 # TODO: Now Spresense environment is not support tests.
 #       So remove every tests.
 MICROLITE_TEST_SRCS := 
-MICRO_LITE_EXAMPLE_TESTS :=
+# MICRO_LITE_EXAMPLE_TESTS :=
 
 # These are microcontroller-specific rules for converting the ELF output
 # of the linker into a binary image that can be loaded directly.
