--- client.c	2016-10-02 06:27:51.000000000 +0900
+++ client.c	2021-04-15 21:11:26.941699494 +0900
@@ -64,14 +64,20 @@
      memset(&sa,0,sizeof(sa));     
      sa.sa_handler=SIG_IGN;
      sa.sa_flags = SA_NOCLDWAIT;
+#ifndef VTUN_NUTTX
      sigaction(SIGHUP,&sa,NULL);
      sigaction(SIGQUIT,&sa,NULL);
+#endif
      sigaction(SIGPIPE,&sa,NULL);
      sigaction(SIGCHLD,&sa,NULL);
 
      sa.sa_handler=sig_term;
+#ifdef SIGTERM
      sigaction(SIGTERM,&sa,NULL);
+#endif
+#ifdef SIGINT
      sigaction(SIGINT,&sa,NULL);
+#endif
  
      client_term = 0; reconnect = 0;
      while( (!client_term) || (client_term == VTUN_SIG_HUP) ){
