--- linkfd.c	2016-10-02 06:27:51.000000000 +0900
+++ linkfd.c	2021-04-15 21:13:05.635689067 +0900
@@ -201,7 +201,7 @@
 	stat_timer = VTUN_STAT_IVAL;
      }
 
-     if ( ka_timer*stat_timer ){
+     if ( ka_timer && stat_timer ){
        alarm( (ka_timer < stat_timer) ? ka_timer : stat_timer );
      } else {
        alarm( (ka_timer) ? ka_timer : stat_timer );
@@ -362,7 +362,11 @@
 /* Link remote and local file descriptors */ 
 int linkfd(struct vtun_host *host)
 {
+#ifdef VTUN_NUTTX
+     struct sigaction sa, sa_oldint;
+#else
      struct sigaction sa, sa_oldterm, sa_oldint, sa_oldhup;
+#endif
      int old_prio;
 
      lfd_host = host;
@@ -377,12 +381,13 @@
      if(host->flags & VTUN_LZO)
 	lfd_add_mod(&lfd_lzo);
 
-     if(host->flags & VTUN_ENCRYPT)
+     if(host->flags & VTUN_ENCRYPT) {
        if(host->cipher == VTUN_LEGACY_ENCRYPT) {
 	 lfd_add_mod(&lfd_legacy_encrypt);
        } else {
 	 lfd_add_mod(&lfd_encrypt);
        }
+     }
      
      if(host->flags & VTUN_SHAPE)
 	lfd_add_mod(&lfd_shaper);
@@ -392,10 +397,16 @@
 
      memset(&sa, 0, sizeof(sa));
      sa.sa_handler=sig_term;
+#ifdef SIGTERM
      sigaction(SIGTERM,&sa,&sa_oldterm);
+#endif
+#ifdef SIGINT
      sigaction(SIGINT,&sa,&sa_oldint);
+#endif
      sa.sa_handler=sig_hup;
+#ifndef VTUN_NUTTX
      sigaction(SIGHUP,&sa,&sa_oldhup);
+#endif
 
      /* Initialize keep-alive timer */
      if( host->flags & (VTUN_STAT|VTUN_KEEP_ALIVE) ){
@@ -434,9 +445,15 @@
 
      lfd_free_mod();
      
+#ifdef SIGTERM
      sigaction(SIGTERM,&sa_oldterm,NULL);
+#endif
+#ifdef SIGINT
      sigaction(SIGINT,&sa_oldint,NULL);
+#endif
+#ifndef VTUN_NUTTX
      sigaction(SIGHUP,&sa_oldhup,NULL);
+#endif
 
      setpriority(PRIO_PROCESS,0,old_prio);
 
