diff -upr Src/main.c Src/main.c
--- Src/main.c	2021-04-27 11:20:28.299280000 +0900
+++ Src/main.c	2021-04-20 11:09:22.000000000 +0900
@@ -25,7 +25,7 @@
 
 /* Private includes ----------------------------------------------------------*/
 /* USER CODE BEGIN Includes */
-
+#include "main_LPWA_sample_app.h"
 /* USER CODE END Includes */
 
 /* Private typedef -----------------------------------------------------------*/
@@ -91,7 +91,7 @@ int main(void)
   MX_USART2_UART_Init();
   MX_RTC_Init();
   /* USER CODE BEGIN 2 */
-
+  main_LPWA_sample_app();
   /* USER CODE END 2 */
 
   /* Infinite loop */
Only in Src: main_LPWA_sample_app.c
diff -upr Src/stm32l0xx_it.c Src/stm32l0xx_it.c
--- Src/stm32l0xx_it.c	2021-04-27 11:20:27.862447400 +0900
+++ Src/stm32l0xx_it.c	2021-04-20 11:09:22.000000000 +0900
@@ -52,7 +52,8 @@
 
 /* Private user code ---------------------------------------------------------*/
 /* USER CODE BEGIN 0 */
-
+#include "CXM150x_Port.h"
+#include "main_LPWA_sample_app.h"
 /* USER CODE END 0 */
 
 /* External variables --------------------------------------------------------*/
@@ -146,7 +147,7 @@ void SysTick_Handler(void)
 void RTC_IRQHandler(void)
 {
   /* USER CODE BEGIN RTC_IRQn 0 */
-
+  rtc_callback();
   /* USER CODE END RTC_IRQn 0 */
   HAL_RTC_AlarmIRQHandler(&hrtc);
   /* USER CODE BEGIN RTC_IRQn 1 */
@@ -160,7 +161,7 @@ void RTC_IRQHandler(void)
 void EXTI0_1_IRQHandler(void)
 {
   /* USER CODE BEGIN EXTI0_1_IRQn 0 */
-
+  wrapper_CXM150x_int_out1();
   /* USER CODE END EXTI0_1_IRQn 0 */
   HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_0);
   /* USER CODE BEGIN EXTI0_1_IRQn 1 */
@@ -174,7 +175,12 @@ void EXTI0_1_IRQHandler(void)
 void EXTI4_15_IRQHandler(void)
 {
   /* USER CODE BEGIN EXTI4_15_IRQn 0 */
-
+    if(__HAL_GPIO_EXTI_GET_IT(GPIO_PIN_10) != RESET){
+        wrapper_CXM150x_int_out2();
+    }
+    if(__HAL_GPIO_EXTI_GET_IT(GPIO_PIN_13) != RESET){
+        push_btn_CXM150x();
+    }
   /* USER CODE END EXTI4_15_IRQn 0 */
   HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_10);
   HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_13);
diff -upr Src/usart.c Src/usart.c
--- Src/usart.c	2021-04-27 11:20:27.743765600 +0900
+++ Src/usart.c	2021-04-20 11:09:22.000000000 +0900
@@ -21,7 +21,7 @@
 #include "usart.h"
 
 /* USER CODE BEGIN 0 */
-
+extern void wrapper_CXM150x_uart_rx_callback(void);
 /* USER CODE END 0 */
 
 UART_HandleTypeDef huart1;
@@ -185,6 +185,11 @@ void HAL_UART_MspDeInit(UART_HandleTypeD
 }
 
 /* USER CODE BEGIN 1 */
+void HAL_UART_RxCpltCallback(UART_HandleTypeDef *UartHandle){
+    if(UartHandle->Instance==USART1){
+        wrapper_CXM150x_uart_rx_callback();
+    }
+}
 
 /* USER CODE END 1 */
 
